### PR TITLE
Fix SA1602/S6608/S4144 warnings

### PR DESCRIPTION
**Pull request by gintsk**
## The issue or feature being addressed
Contributes to #1290

## Details on the issue fix or feature implementation
Fix SA1602
Fix S6608 (net481 style) 
Fix S4144 (remove duplicate tests)

## Confirm the following

- [x]  I started this PR by branching from the head of the default branch
- [x]  I have targeted the PR to merge into the default branch
- [x]  I have included unit tests for the issue/feature
- [x]  I have successfully run a local build

### CODE REVIEW

**Comment by gintsk** [comment]
@dotnet-policy-service agree

File: test/Polly.Core.Tests/Retry/RetryResilienceStrategyTests.cs [hunk 1]
```diff
@@ -89,9 +89,9 @@ public void ExecuteAsync_MultipleRetries_EnsureDiscardedResultsDisposed()
         // assert
         result.IsDisposed.Should().BeFalse();
         results.Count.Should().Be(_options.MaxRetryAttempts + 1);
-        results.Last().IsDisposed.Should().BeFalse();
+        results[results.Count - 1].IsDisposed.Should().BeFalse();
 
-        results.Remove(results.Last());
+        results.Remove(results[results.Count - 1]);
```
Comment (martincostello) [inline test/Polly.Core.Tests/Retry/RetryResilienceStrategyTests.cs:94]: Does using the terser `[^1]` work here (and in the other files) instead?

File: test/Polly.Specs/PolicyTResultAsyncSpecs.cs [hunk 2]
```diff
@@ -110,18 +110,6 @@ public async Task Executing_the_policy_function_should_throw_when_context_is_nul
...
-    public async Task Execute_and_capturing_the_policy_function_should_throw_when_context_data_is_null()
-    {
-        var policy = Policy
-            .HandleResult(ResultPrimitive.Fault)
-            .RetryAsync((_, _, _) => { });
-
-        var ex = await policy.Awaiting(p => p.ExecuteAndCaptureAsync(_ => Task.FromResult(ResultPrimitive.Good), null!))
-              .Should().ThrowAsync<ArgumentNullException>();
-        ex.And.ParamName.Should().Be("context");
-    }
```
Comment (martincostello) [inline test/Polly.Specs/PolicyTResultAsyncSpecs.cs:123]: This isn't a duplicate of the method above as it's calling `ExecuteAndCaptureAsync()` not `ExecuteAsync()`.

File: test/Polly.Core.Tests/Retry/RetryResilienceStrategyTests.cs (hunk 1 above)
Comment (gintsk) [reply test/Polly.Core.Tests/Retry/RetryResilienceStrategyTests.cs:94]: Unfortunately `[^1]` can not be used due net481

File: test/Polly.Specs/PolicyTResultAsyncSpecs.cs (hunk 2 above)
Comment (gintsk) [reply test/Polly.Specs/PolicyTResultAsyncSpecs.cs:123]: The duplicates are not next to each other 
The deleted one
`public async Task Execute_and_capturing_the_policy_function_should_throw_when_context_data_is_null()`
duplicates this one
`public async Task Execute_and_capturing_the_policy_function_should_throw_when_context_is_null()`

**Review by martincostello — APPROVED** [review]
Thanks!

**Pull request merged**